internal/app/command.go

```diff
@@ -69,33 +56,58 @@ func (c *Command) Exec() ExitStatus {
 	cmd.Stderr = &stderr
 
 	if err := cmd.Start(); err != nil {
-		log.Info("cmd.Start: " + err.Error())
-		return ExitStatus{
-			code:   1,
-			errors: err.Error(),
-		}
+		return nil, err
 	}
 
 	if err := cmd.Wait(); err != nil {
 		if exiterr, ok := err.(*exec.ExitError); ok {
-			if status, ok := exiterr.Sys().(syscall.WaitStatus); ok {
-				return ExitStatus{
-					code:   status.ExitStatus(),
-					output: stdout.String(),
-					errors: stderr.String(),
-				}
-			}
+			return nil, newExitError(c, exiterr.ExitCode(), stdout, stderr)
 		} else {
 			log.Fatal("cmd.Wait: " + err.Error())
 		}
 	}
-	return ExitStatus{
-		code:   0,
-		output: stdout.String(),
-		errors: stderr.String(),
+	return newExitStatus(c, stdout, stderr), nil
+}
+
+type ExitStatus struct {
+	cmd    string
+	output string
+	errors string
+}
+
+func newExitStatus(cmd *Command, stdout, stderr bytes.Buffer) *ExitStatus {
+	return &ExitStatus{
+		cmd:    cmd.String(),
+		output: strings.TrimSpace(stdout.String()),
+		errors: strings.TrimSpace(stderr.String()),
 	}
 }
 
+func (es *ExitStatus) String() string {
+	return fmt.Sprintf("cmd %s exited successfully\noutput: %s", es.cmd, es.output)
+}
+
+type exitError struct {
+	cmd    string
+	code   int
+	output string
+}
+
+func newExitError(cmd *Command, code int, stdout, stderr bytes.Buffer) *exitError {
+	return &exitError{
+		cmd:  cmd.String(),
+		code: code,
+		output: fmt.Sprintf(
+			"--- stdout ---\n%s\n--- stderr ---\n%s",
+			strings.TrimSpace(stdout.String()),
+			strings.TrimSpace(stderr.String())),
+	}
+}
+
+func (ee *exitError) Error() string {
+	return fmt.Sprintf("cmd %s failed with non-zero exit code %d\noutput: %s", ee.cmd, ee.code, ee.output)
+}
```

Comment:
if you're providing a custom error type why not export it so it can be used to retrieve the error details? Otherwise this seems overkill and all we really need is `fmt.Errorf`, this is what I meant on my comment in the issue.